pkg/operation/botanist/component/extensions/worker/worker.go

```diff
@@ -281,15 +286,20 @@ func (w *worker) Destroy(ctx context.Context) error {
 
 // Wait waits until the Worker resource is ready.
 func (w *worker) Wait(ctx context.Context) error {
+	waitTimeout, severeThreshold, interval, err := w.DetermineWaitTimeout(ctx)
+	if err != nil {
+		return err
+	}
+
 	return extensions.WaitUntilExtensionObjectReady(
 		ctx,
 		w.client,
 		w.logger,
 		w.worker,
 		extensionsv1alpha1.WorkerResource,
-		w.waitInterval,
-		w.waitSevereThreshold,
-		w.waitTimeout,
+		interval,
+		severeThreshold,
+		waitTimeout,
```

Comment:
This means that we block an entire worker for the maximum drain timeout, right? I think we should be careful with that even if it's understandable that reconciling again and again isn't nice either.

Reply:
Correct, that goroutine will be blocked during that time.
At least to me, that does not seem to be very problematic:
- low amounts of Shoots with custom drain timeouts
- few Shoots reconciling usually at the same time

The probability of many Shoots with custom drain timeouts reconciling at the same time + being stuck at the same time is pretty low. If we run into issues, we could also increase the workers (+ CPU limit).

Reply:
Would be nice to have a second opinion here

Reply:
I was more thinking about a potential DOS attack vector (may be very strict but still) and it even further contradicts with the short reconcile best practice. Actually, it'd be nice to have a proper re-enqueuing mechanism that is time based.
/cc @timebertt

Reply:
Ah, that is where you are coming from. Valid concerns.

Reply:
I would second @timuthy's comment, blocking an entire goroutine for that long is not what we should do. The today's wait mechanisms are already violating best practices, but if we now change this to even hours or days it can just get worse. Regularly requeueing seems to be the better approach.

Reply:
I can only agree with what @timuthy and @rfranzke already said. That's exactly my position on this as well.

Ideally, there doesn't need to be any `wait` or `time.Sleep` in k8s controllers (and there shouldn't be because of good reasons – think scalability, etc.).
Especially extensions have it pretty easy to watch the necessary `MachineDeployment`, `MachineSet`, `Machine` objects and enqueue the `Worker` whenever something relevant changes.

Reply:
Thanks for your input. I removed the changes.